pkgs/development/interpreters/lua-5/tests/default.nix

```
@@ -18,29 +15,43 @@ let
     + "touch $out"
     );
 
-    wrappedHello = hello.overrideAttrs(oa: {
-      propagatedBuildInputs = [
-        wrapLua
-        lua.pkgs.cjson
-      ];
-      postFixup = ''
-        wrapLuaPrograms
-      '';
-    });
+  wrappedHello = hello.overrideAttrs(oa: {
+    propagatedBuildInputs = [
+      wrapLua
+      lua.pkgs.cjson
+    ];
+    postFixup = ''
+      wrapLuaPrograms
+    '';
+  });
+
+  luaWithModule = lua.withPackages(ps: [
+    ps.lua-cjson
+  ]);
 
-    luaWithModule = lua.withPackages(ps: [
-      ps.lua-cjson
-    ]);
+  golden_LUA_PATHS = {
+
+    "5.1" = "./?.lua;${lua}/share/lua/5.1/?.lua;${lua}/share/lua/5.1/?/init.lua;${lua}/lib/lua/5.1/?.lua;${lua}/lib/lua/5.1/?/init.lua";
+    "5.2" = "${lua}/share/lua/5.2/?.lua;${lua}/share/lua/5.2/?/init.lua;${lua}/lib/lua/5.2/?.lua;${lua}/lib/lua/5.2/?/init.lua;./?.lua";
+    "5.3" = "${lua}/share/lua/5.3/?.lua;${lua}/share/lua/5.3/?/init.lua;${lua}/lib/lua/5.3/?.lua;${lua}/lib/lua/5.3/?/init.lua;./?.lua;./?/init.lua";
+    "5.4" = "${lua}/share/lua/5.4/?.lua;${lua}/share/lua/5.4/?/init.lua;${lua}/lib/lua/5.4/?.lua;${lua}/lib/lua/5.4/?/init.lua;./?.lua;./?/init.lua";
+    "luajit" = "./?.lua;${lua}/share/luajit-2.1/?.lua;/usr/local/share/lua/5.1/?.lua;/usr/local/share/lua/5.1/?/init.lua;${lua}/share/lua/5.1/?.lua;${lua}/share/lua/5.1/?/init.lua";
+  };
 in
   pkgs.recurseIntoAttrs ({
 
-  checkAliases = runTest lua {
-    name = "check-aliases";
+  checkInterpreterPatch = let
+    golden_LUA_PATH = if lua.pkgs.luaLib.isLuaJIT then golden_LUA_PATHS."luajit" else golden_LUA_PATHS.${lua.luaversion};
+  in
+    runTest lua {
+    name = "check-default-lua-path";
     command = ''
       generated=$(lua -e 'print(package.path)')
-      golden_LUA_PATH='./share/lua/${lua.luaversion}/?.lua;./?.lua;./?/init.lua'
-
-      assertStringContains "$generated" "$golden_LUA_PATH"
+      echo "lua: ${lua}"
+      echo "${golden_LUA_PATH}"
+      assertStringEqual "$generated" "${golden_LUA_PATH}"
+      export LUA_PATH=";;"
+      assertStringEqual "$generated" "${golden_LUA_PATH}"
```

```suggestion
      assertStringEqual "$generated" ";${golden_LUA_PATH};"
```

`;;` actually gets replaced with `;LUA_PATH_DEFAULT;` to make sure paths don't get blended together.

---

well spotted ! This makes sense. Rebasing on your fix showed the test was broken.

I've updated the commit as you suggested with this additional commit message:
```
    when LUA_PATH contains `;;`, it is replaced by `;LUA_PATH_DEFAULT;` in
    lua 5.1:
    Updated https://github.com/lua/lua/blob/69ea087dff1daba25a2000dfb8f1883c17545b7a/loadlib.c#L599
    More recent versions might be smarter ?
    https://github.com/lua/lua/blob/65b07dd53d7938a60112fc4473f5cad3473e3534/loadlib.c#L301
```

---

hmm, if it differs per-version that might make the tests significantly more complicated.  maybe we could just strip leading and trailing `;` in lua via gsub?  we'll see what ofborg says first.